clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -464,6 +464,8 @@ private void handleGroupMetadataResponse(ClientResponse resp, RequestFuture<Void
                         groupMetadataResponse.node().host(),
                         groupMetadataResponse.node().port());
 
+                client.tryConnect(coordinator);
```

Comment:
It's to ensure that the connection doesn't stay in the DISCONNECTED state indefinitely. The only way to break out of that state is to call NetworkClient.ready() (tryConnect delegates to NetworkCleint.ready()). Otherwise, the coordinatorUnknown() check in ensureCoordinatorKnown() would always return true after the connection had failed.